source/eventcore/core.d

```diff
@@ -27,24 +27,32 @@ else alias NativeEventDriver = EventDriver;
 static if (!is(NativeEventDriver == EventDriver)) {
 	static this()
 	{
-		if (!s_driver) s_driver = new NativeEventDriver;
+		if (!s_initCount++) {
+			if (!s_driver) s_driver = new NativeEventDriver;
+		}
 	}
 
 	static ~this()
 	{
-		if (!s_isMainThread)
-			s_driver.dispose();
+		if (!s_isMainThread) {
+			if (!--s_initCount)
+				s_driver.dispose();
+		}
```

Comment:
I am not sure this works:

Consider two threads:

```
shared static this() @ T1 -> s_initCount: 1
static this() @ T1 -> s_initCount: 2
static this() @ T2 -> s_initCount2: 1
~static this() @ T1 -> s_initCount: 2 (<-no decrement!)
~static this() @ T2 -> s_initCount2: 0
~shared static this() @ T1 -> s_initCount: 1
```

Reply:
True indeed, `static this` actually needs to check `s_isMainThread ` the same way...